app/routes.py

```diff
@@ -1,2 +1,239 @@
+        completed_at = request_body["completed_at"]
+        )
+    db.session.add(new_task)
+    db.session.commit()
+    
+    response_body = {"task": {"id": new_task.id,
+        "title": new_task.title,
+        "description": new_task.description,
+        "is_complete": bool(new_task.completed_at)}}
+        
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])   
+def get_tasks():
+        sort_query = request.args.get("sort")
+        if sort_query == "asc":
+            tasks = Task.query.order_by(Task.title.asc()).all()
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        else:
+            tasks = Task.query.all()
+
+        tasks_response = []
+        for task in tasks: 
+            tasks_response.append({
+                "id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)
+                })
+
+        return jsonify(tasks_response), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if not task:
+        return make_response(f"Task {task_id} not found", 404)
+        
+    if request.method == "GET":
+        if not task.goal_id:
+            response_body = {"task":
+                {"id": task.id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+        
+        else:
+            response_body = {"task":
+            {"id": task.id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+            return jsonify(response_body), 200
+
+    elif request.method == "PUT":
+        request_body = request.get_json()
+        task.title = request_body['title']
+        task.description = request_body['description']
+        if 'completed_at' in request_body:
+            task.is_complete = True
+        else:
+            task.is_complete = False
+        db.session.commit()
+        
+        response_body = {"task": {"id": task.id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": bool(task.completed_at)}}
+        
+        return jsonify(response_body), 200
+
+    elif request.method == "DELETE":
+        db.session.delete(task)
+        db.session.commit()
+        return {"details":(f'Task {task.id} "{task.title}" successfully deleted')}, 200
+
+
+def slack_bot(title):
+    query_path = {'channel':'slack_api_test_channel', 'text': title}
+    header = {'Authorization': os.environ.get('BOT')}
+    response=requests.post('https://slack.com/api/chat.postMessage', params = query_path, headers= header)
```

Comment:
For the `post` request, notice that when we make a post request, we tend not to pass parameters as part of the query params. Instead, we usually use the request body. Slack supports JSON body content, which we can get by changing the `params` named parameter in the `post` call to `json`.